controllers/reconcile.go

```diff
@@ -1359,6 +1359,42 @@ func createConfigmapHash(cfgMap *v1.ConfigMap) (string, error) {
 	return hash, nil
 }
 
+func setupAPISpecs(feList *crd.FrontendList) []crd.APISpecInfo {
+	var allSpecs []crd.APISpecInfo
+
+	for _, frontend := range feList.Items {
+		if frontend.Spec.API != nil && len(frontend.Spec.API.Specs) > 0 {
+			allSpecs = append(allSpecs, frontend.Spec.API.Specs...)
+		}
+	}
+
+	// Sort deterministically by ServiceRef then URL
+	sort.Slice(allSpecs, func(i, j int) bool {
+		serviceRefI := allSpecs[i].ServiceRef
```

Comment:
@justinorringer it does not appear that `ServiceRef` is being set anywhere currently? Should the operator be populating it internally before attempting this sorting?